graphs/uniform_search_cost.py

```diff
@@ -0,0 +1,181 @@
+##################### -- Learning Source -- #####################
+
+# 1. https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm
+# 2. https://www.educative.io/answers/what-is-uniform-cost-search
+
+# In above 1 no. link, you will get about dijkstra algorithm.
+# You may have to scroll down a little bit.
+# And you will get a section titled as "Practical optimizations and infinite graphs"
+# This section explains about Uniform Cost Search algorithm.
+
+# Uniform search cost is slightly similar to dijkstra.
+# The main different between them is that UCS is having multiple goals.
+
+# Below code is fully based on graph node environment
+##################### --------------------- #####################
+
+
+def uniform_search_cost(
+    graph_data: list[list[int]],
+    distance: dict[
+        tuple[
+            int,
+            int,
+        ],
+        int,
+    ],
+    goal: list[int],
+    start: int,
+) -> list[int]:
+    """
+    Returns a list of integer values which are result of optimal distance
+    >>> graph_data = [[1, 2], [2, 3], [], []]
+    >>> distance = {(0, 1): 2, (0, 2): 1, (1, 2): 5, (1, 3): 10}
+    >>> goal = [2]
+    >>> start = 0
+    >>> uniform_search_cost(graph_data, distance, goal, start)
+    [1]
+
+    >>> graph_data = [[1, 2], [2, 3], [], [0, 2]]
+    >>> distance = {(0,1):2.91,(0,2):1.1,(1,2):3,(1,3):2.3,(3,0):2.34,(3,2):2.15}
+    >>> goal = [2]
+    >>> start = 0
+    >>> uniform_search_cost(graph_data, distance, goal, start)
+    [1.1]
+
+    >>> graph_data = [[1, 2], [2, 3], [], [0, 2]]
+    >>> distance = {(0,1):2.91,(0,2):0,(1,2):3,(1,3):2.3,(3,0):2.34,(3,2):2.15}
+    >>> goal = [2]
+    >>> start = 0
+    >>> uniform_search_cost(graph_data, distance, goal, start)
+    [0]
+
+    >>> graph_data = [[1, 2], [2, 3], [], [0, 2]]
+    >>> distance = {(0,1):1.3,(0,2):2.6,(1,2):0,(1,3):2.3,(3,0):2.34,(3,2):2.15}
+    >>> goal = [2]
+    >>> start = 0
+    >>> uniform_search_cost(graph_data, distance, goal, start)
+    [1.3]
+    """
+    # priority queue
+    heap = []
+    # Goal result to be max value initially
+    goal_answer = [10**8] * len(goal)
```

Comment:
Please use something like `float("inf")` for the arbitrary max value. `10**8` wouldn't work as a max value if, for example, an edge has distance greater than 10^8.